### PR TITLE
Add convenience installation script

### PR DESCRIPTION
**Pull request by fruttasecca**
## Description

Fixes: #951

- [x] post https://github.com/orchest/orchest/pull/966 adjustments

### CODE REVIEW

**Comment by ricklamers** [comment]
Can we detect `pip` and if it is not installed show recommended installation process?

Did you decide to scope support to macOS and Linux?

Do we point Windows users somewhere (`Git Bash`, `Cygwin`, `MinGW`, `WSL2`)?

**Comment by fruttasecca** [comment]
>Can we detect pip and if it is not installed show a link to recommended installation process?

Sure

>Did you decide to scope support to macOS and Linux?
>Do we point Windows users somewhere (Git Bash, Cygwin, MinGW, WSL2)?

macOS was an easy increment from a linux only script, I am not sure if it makes sense to cover windows users in any form unless we want to increase the scope of the PR to a doc that explains everything you mentioned, which would actually be about "how to install and use Orchest on windows", which was something we forego AFAIR (the docs don't mention windows anymore)

**Comment by ricklamers** [comment]
Windows can still work, iff they use a `bash` shell. So I think it's good like this! You can merge imo. I'll do some more testing for edge cases in the future, but I think this is a good enough start.

**Review by ricklamers — APPROVED** [review]

File: docs/source/getting_started/installation.rst [hunk 1]
```diff
@@ -6,6 +6,10 @@ Installation
 .. note::
    Orchest is in beta.
 
+.. tip::
+   👉 A convenience end to end installation script for a minikube deployment is provided
+   in ``scripts/convenience_install.sh``.
```
[the remark this answers is not in the document]
Comment (fruttasecca) [reply docs/source/getting_started/installation.rst:11]: Sure

[the remark this answers is not in the document]
Comment (fruttasecca) [reply docs/source/getting_started/installation.rst]: https://github.com/orchest/orchest/pull/983/commits/0e7c7553badfa6b1fb710aea68482e3faf9b14a4

**Comment by fruttasecca** [comment]
Can't be tested end to end until the controller PR is merged

File: scripts/convenience_install.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+fi
+
+if ! minikube addons list | grep "ingress " | grep "enabled" > /dev/null ; then
+    echo "Enabling ingress addon..."
+    minikube addons enable ingress
+fi
+
+pip install orchest-cli --upgrade
+
+orchest install
```
Comment (yannickperrenet) [inline scripts/convenience_install.sh:50]: How about wrapping it in an if-statement checking the exit code of `orchest version --json`?

Comment (yannickperrenet) [reply scripts/convenience_install.sh:50]: Yes. But we could just let `orchest install` handle that case of course (which it already does). Might even be better :thinking:

Let's leave it as is :+1:
